Makefile

```diff
@@ -174,7 +174,7 @@ release-binary: $(RELEASE_DIR)
 		-v "$$(pwd):/workspace$(DOCKER_VOL_OPTS)" \
 		-w /workspace/tools/setup-envtest \
 		golang:$(GO_VERSION) \
-		go build -a -trimpath -ldflags "-extldflags '-static'" \
+		go build -a -trimpath -ldflags "-X 'main.BranchVersion=$(RELEASE_TAG)' -extldflags '-static'" \
```

Comment:
Let's call this version instead of BranchVersion

I assume this works, do you know if we can also use something more like in controller-gen?

e.g. sigs.k8s.io/controller-runtime/tools/setup-envtest/main.version=

Maybe also move the const to a tools/setup-envtest/version package, but not sure what the best practices are there